scylla/src/transport/iterator.rs

```diff
@@ -146,16 +146,24 @@ impl RowIterator {
 
         tokio::task::spawn(worker_task);
 
-        RowIterator {
+        let pages_received = receiver.recv().await.unwrap()?;
```

Comment:
It's not really obvious if the worker always pushes at least one value before the channel is closed. If not, the `unwrap()` here will trigger a panic which will crash the application - and that is something we would like to avoid.

Let's be defensive and use `unwrap_with_default()` instead - it will return a `Default::default()` page if the channel does not produce any page.

Additionally, the variable holds only one page, so the page in the variable name should be singular (`page_received`).

The same comments apply to the logic in `new_for_prepared_statement`.

Reply:
After our conversation in private I was convinced that unwrapping here is OK. It's a bug if there are no pages returned.